File: src/jbased/LayerDense.java
package jbased;

class LayerDense {
    double[][] weights;
    double[][] biases;
    double[][] output;

    public LayerDense(int n_inputs, int n_neurons) {
        weights = Part4.generateMatrix(n_inputs, n_neurons);
        biases = new double[n_neurons][];

    }

    public void forward(double[][] inputs) {
        output=Part3.dotProduct(inputs, weights);
    }
}

File: src/kbased/LayerDense.kt
package kbased

import generateMatrix


internal class LayerDense(n_inputs: Int, n_neurons: Int) {
    var weights: Array<DoubleArray>
    var biases: Array<DoubleArray?>
    lateinit var output: Array<DoubleArray>
    fun forward(inputs: Array<DoubleArray>) {
        output = dotProduct(inputs, weights)
    }

    init {
        weights = generateMatrix(n_inputs, n_neurons)
        biases = arrayOfNulls(n_neurons)
    }
}